nuxeo-runtime/nuxeo-runtime-jtajca/src/main/java/org/nuxeo/runtime/jtajca/NuxeoConnectionManager.java

```diff
@@ -0,0 +1,402 @@
+     * @since 8.4
+     */
+    public long getKilledConnectionCount() {
+        return activemonitor.killedCount;
+    }
+
+    final ActiveMonitor activemonitor;
+
+    class ActiveMonitor implements ConnectionTracker {
+
+        final int ttl;
+
+        ActiveMonitor(int delay) {
+            ttl = delay;
+            if (ttl > 0) {
+                scheduleCleanups();
+            }
+            coordinator.addTracker(this);
+        }
+
+        final Map<ConnectionInfo, TimeToLive> ttls = new HashMap<>();
+
+        long killedCount = 0L;
+
+        CleanupTask cleanup = new CleanupTask();
+
+        class CleanupTask extends TimerTask {
+
+            @Override
+            public void run() {
+                killActiveTimedoutConnections(System.currentTimeMillis());
+            }
+
+        }
+
+        void cancelCleanups() {
+            cleanup.cancel();
+        }
+
+        void scheduleCleanups() {
+            PoolIdleReleaserTimer.getTimer().scheduleAtFixedRate(cleanup, 60 * 1000, 60 * 1000);
+        }
+
+        @Override
+        public synchronized void handleObtained(ConnectionTrackingInterceptor connectionTrackingInterceptor,
+                ConnectionInfo connectionInfo, boolean reassociate) throws ResourceException {
+            int delay = ttl();
+            if (delay > 0) {
+                ttls.put(connectionInfo, new TimeToLive(connectionInfo, Thread.currentThread().getName(), System.currentTimeMillis(), delay));
+            }
+        }
+
+        @Override
+        public synchronized void handleReleased(ConnectionTrackingInterceptor connectionTrackingInterceptor,
+                ConnectionInfo connectionInfo, ConnectionReturnAction connectionReturnAction) {
+            ttls.remove(connectionInfo);
+        }
+
+        @Override
+        public void setEnvironment(ConnectionInfo connectionInfo, String key) {
+            return;
+        }
+
+        /**
+         * Kill active connection that have timed out relative to the given {@code clock}.
+         *
+         * @return information about the killed connections
+         * @param clock
+         * @since 8.4
+         */
+        public synchronized List<TimeToLive> killTimedoutConnections(long clock) {
+            List<TimeToLive> killed = new LinkedList<>();
+            Iterator<TimeToLive> it = ttls.values().iterator();
+            while (it.hasNext()) {
+                TimeToLive ttl = it.next();
+                if (ttl.deadline <= clock) {
+                    ttl.killAndLog();
+                    killed.add(ttl);
+                    it.remove();
+                }
+
+            }
+            return killed;
+        }
+
+        /**
+         * Logs active connections
+         *
+         *
+         * @since 8.4
+         */
+        public void log() {
+            for (TimeToLive ttl : ttls.values()) {
+                LogFactory.getLog(TimeToLive.class).warn(ttl, ttl.info.getTrace());
```

Comment:
Maybe we should rely on a log method in TimeToLive class which declares a logger.